src/react/hooks/__tests__/useLazyQuery.test.tsx

```diff
@@ -1483,6 +1483,83 @@ describe("useLazyQuery Hook", () => {
     expect(fetchCount).toBe(1);
   });
 
+  // https://github.com/apollographql/apollo-client/issues/9448
+  it.each(["network-only", "no-cache", "cache-and-network"] as const)(
+    "does not issue multiple network calls when calling execute again without variables with a %s fetch policy",
+    async (fetchPolicy) => {
+      interface Data {
+        user: { id: string; name: string };
+      }
+
+      interface Variables {
+        id?: string;
+      }
+
+      const query: TypedDocumentNode<Data, Variables> = gql`
+        query UserQuery($id: ID) {
+          user(id: $id) {
+            id
+            name
+          }
+        }
+      `;
+
+      let fetchCount = 0;
+
+      const link = new ApolloLink((operation) => {
+        fetchCount++;
+        return new Observable((observer) => {
+          setTimeout(() => {
+            observer.next({
+              data: {
+                user: { id: operation.variables.id || null, name: "John Doe" },
+              },
+            });
+            observer.complete();
+          }, 20);
+        });
+      });
+
+      const client = new ApolloClient({
+        link,
+        cache: new InMemoryCache(),
+      });
+
+      const { result } = renderHook(
+        () => useLazyQuery(query, { fetchPolicy }),
+        {
+          wrapper: ({ children }) => (
+            <ApolloProvider client={client}>{children}</ApolloProvider>
+          ),
+        }
+      );
+
+      const [execute] = result.current;
+
+      await act(() => execute({ variables: { id: "2" } }));
```

Comment:
Let's not destructure `execute` here - this might hide some bugs where a call with a stale version of `execute` might behave differenly from a call with a non-stale version.

```suggestion
      await act(() => result.current.execute({ variables: { id: "2" } }));
```

Reply:
Done! [b2b61a1](https://github.com/apollographql/apollo-client/pull/11403/commits/b2b61a19f999503679ed7014b4068a6db2832bae)